src/components/Reports/DesignationAccountsReport/DesignationAccountsReport.tsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { useTranslation } from 'react-i18next';
+import { Box, CircularProgress, Divider, styled } from '@material-ui/core';
+import { useDesignationAccountsQuery } from './GetDesignationAccounts.generated';
+import { useSetActiveDesignationAccountMutation } from './SetActiveDesignationAccount.generated';
+import { DesignationAccountsHeader as Header } from './Layout/Header/Header';
+import { DesignationAccountsList as List } from './Layout/List/List';
+import { Notification } from 'src/components/Notification/Notification';
+import { EmptyReport } from 'src/components/Reports/EmptyReport/EmptyReport';
+import { currencyFormat } from 'src/lib/intlFormat';
+
+interface Props {
+  accountListId: string;
+  isNavListOpen: boolean;
+  onNavListToggle: () => void;
+  title: string;
+}
+
+const ScrollBox = styled(Box)(({}) => ({
+  height: 'calc(100vh - 160px)',
+  overflowY: 'auto',
+}));
+
+export const DesignationAccountsReport: React.FC<Props> = ({
+  accountListId,
+  isNavListOpen,
+  onNavListToggle,
+  title,
+}) => {
+  const { t } = useTranslation();
+
+  const { data, loading, error, refetch } = useDesignationAccountsQuery({
+    variables: {
+      accountListId,
+    },
+  });
+
+  const [
+    setActiveDesignationAccount,
+  ] = useSetActiveDesignationAccountMutation();
+
+  const handleCheckToggle = async (
+    event: React.ChangeEvent<HTMLInputElement>,
+    designationAccountId: string,
+  ) => {
+    await setActiveDesignationAccount({
+      variables: {
+        input: {
+          accountListId,
+          active: event.target.checked,
+          designationAccountId,
+        },
+      },
+    });
+
+    return refetch();
```

Comment:
There should be better ways to do this. See my other comment. Hopefully you can just remove this all together.

If you can get the other suggestion working, might be nice to provide an [optimistic response](https://www.apollographql.com/docs/react/performance/optimistic-ui/#the-optimisticresponse-option) so you don't have to wait for a round trip to the server before the box gets checked.